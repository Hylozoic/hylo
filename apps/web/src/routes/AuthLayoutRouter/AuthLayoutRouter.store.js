import { LOCATION_CHANGE } from 'redux-first-history'
import { pick } from 'lodash/fp'
import rollbar from 'client/rollbar'
import {
  FETCH_FOR_CURRENT_USER,
  FETCH_FOR_GROUP_PENDING,
  LOGOUT_PENDING
} from 'store/constants'

export const MODULE_NAME = 'AuthLayoutRouter'

const TOGGLE_DRAWER = `${MODULE_NAME}/TOGGLE_DRAWER`

const TOGGLE_NAV_MENU = `${MODULE_NAME}/TOGGLE_NAV_MENU`

export const initialState = {
  isDrawerOpen: false,
  isNavOpen: false
}

export default function reducer (state = initialState, action) {
  if (action.error) return state

  if (action.type === TOGGLE_DRAWER) {
    return { ...state, isDrawerOpen: !state.isDrawerOpen }
  }

  if (action.type === TOGGLE_NAV_MENU) {
    return { ...state, isNavOpen: action.value !== undefined ? action.value : !state.isNavOpen }
  }

  if (action.type === LOCATION_CHANGE) {
    return { ...state, isDrawerOpen: false }
  }

  // Links current user to rollbar config
  if (action.type === FETCH_FOR_CURRENT_USER && action.payload?.data?.me) {
    const { id, name, email } = action.payload.data.me
    rollbar.configure({
      payload: {
        person: {
          id,
          username: name,
          email
        }
      }
    })
  }

  return state
}

export function toggleDrawer () {
  return {
    type: TOGGLE_DRAWER
  }
}

export function toggleNavMenu (value) {
  return {
    type: TOGGLE_NAV_MENU,
    value
  }
}

export function ormSessionReducer (
  { Group, Me, Membership, Person },
  { type, meta, payload }
) {
  switch (type) {
    case LOGOUT_PENDING: {
      Me.first().delete()
      break
    }
    case FETCH_FOR_GROUP_PENDING: {
      const group = Group.safeGet({ slug: meta.slug })
      if (!group) return
      const me = Me.first()
      if (!me) return
      const membership = Membership.safeGet({ group: group.id, person: me.id })
      if (!membership) return
      return membership.update({ newPostCount: 0 })
    }
    case FETCH_FOR_CURRENT_USER: {
      const me = payload.data?.me
      if (me && !Person.idExists(me.id)) {
        Person.create(pick(['id', 'name', 'avatarUrl'], me))
      }
      break
    }
  }
}
