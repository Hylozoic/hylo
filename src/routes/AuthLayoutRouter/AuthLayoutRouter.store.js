import { LOCATION_CHANGE } from 'connected-react-router'
import { pick } from 'lodash/fp'
import rollbar from 'client/rollbar'
import {
  FETCH_FOR_CURRENT_USER,
  FETCH_FOR_GROUP_PENDING
} from 'store/constants'

export const MODULE_NAME = 'AuthLayoutRouter'

const TOGGLE_DRAWER = `${MODULE_NAME}/TOGGLE_DRAWER`

const TOGGLE_GROUP_MENU = `${MODULE_NAME}/TOGGLE_GROUP_MENU`

export const initialState = {
  isDrawerOpen: false,
  isGroupMenuOpen: false
}

export default function reducer (state = initialState, action) {
  if (action.error) return state

  if (action.type === TOGGLE_DRAWER) {
    return { ...state, isDrawerOpen: !state.isDrawerOpen }
  }

  if (action.type === TOGGLE_GROUP_MENU) {
    return { ...state, isGroupMenuOpen: !state.isGroupMenuOpen }
  }

  if (action.type === LOCATION_CHANGE) {
    return { ...state, isDrawerOpen: false, isGroupMenuOpen: false }
  }

  // Links current user to rollbar config
  if (action.type === FETCH_FOR_CURRENT_USER) {
    let { id, name, email } = action.payload.data.me
    rollbar.configure({
      payload: {
        person: {
          id: id,
          username: name,
          email: email
        }
      }
    })
  }

  return state
}

export function toggleDrawer () {
  return {
    type: TOGGLE_DRAWER
  }
}

export function toggleGroupMenu () {
  return {
    type: TOGGLE_GROUP_MENU
  }
}

export function ormSessionReducer (
  { Group, Me, Membership, Network, Person },
  { type, meta, payload }
) {
  if (type === FETCH_FOR_GROUP_PENDING) {
    let group = Group.safeGet({ slug: meta.slug })
    if (!group) return
    const me = Me.first()
    if (!me) return
    let membership = Membership.safeGet({ group: group.id, person: me.id })
    if (!membership) return
    membership.update({ newPostCount: 0 })
  }

  if (type === FETCH_FOR_CURRENT_USER) {
    const { me } = payload.data
    if (!Person.idExists(me.id)) {
      Person.create(pick(['id', 'name', 'avatarUrl'], me))
    }
  }
}
