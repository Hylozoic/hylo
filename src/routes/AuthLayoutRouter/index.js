import { withResizeDetector } from 'react-resize-detector'
import component from './AuthLayoutRouter'
import connector from './AuthLayoutRouter.connector'

export default connector(withResizeDetector(component, { handleHeight: false }))
