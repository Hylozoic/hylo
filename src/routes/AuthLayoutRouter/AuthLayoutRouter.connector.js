import { get, some } from 'lodash/fp'
import { connect } from 'react-redux'
import getReturnToPath from 'store/selectors/getReturnToPath'
import setReturnToPath from 'store/actions/setReturnToPath'
import fetchForCurrentUser from 'store/actions/fetchForCurrentUser'
import fetchForGroup from 'store/actions/fetchForGroup'
import updateUserSettings from 'store/actions/updateUserSettings'
import { FETCH_FOR_CURRENT_USER, FETCH_FOR_GROUP } from 'store/constants'
import getMe from 'store/selectors/getMe'
import getGroupForCurrentRoute from 'store/selectors/getGroupForCurrentRoute'
import getMyMemberships from 'store/selectors/getMyMemberships'
import { getSignupInProgress } from 'store/selectors/getSignupState'
import isGroupRoute, { getSlugFromLocation } from 'store/selectors/isGroupRoute'
import { toggleDrawer, toggleGroupMenu } from './AuthLayoutRouter.store'
import getLastViewedGroup from 'store/selectors/getLastViewedGroup'

export function mapStateToProps (state, props) {
  const memberships = getMyMemberships(state, props)
  const showMenuBadge = some(m => m.newPostCount > 0, memberships)
  const hasMemberships = memberships.length > 0
  const slug = getSlugFromLocation(null, props)
  const group = getGroupForCurrentRoute(state, props)
  const lastViewedGroup = getLastViewedGroup(state)
  const currentGroupMembership = group && hasMemberships && memberships.find(m => m.group.id === group.id)
  const signupInProgress = getSignupInProgress(state)
  // NOTE: PENDING as we've generally used it can be misleading and may need reconsideration:
  // Currently the `Pending` state for an action will be `undefined` therefore falsy
  // before a the action has been called. In this case this was causing the component
  // to render through without a `currentUser` on initial render which was causing issues.
  // By default the pendingMiddleware keeps the key once it's been used, so it's value
  // becomes `null`. In most cases I think we want the pending / loading state to be true
  // before and during the initial fetch or action is ran.
  //
  // Recommentaion: Either update `pendingMiddleware` logic, create a selector which
  // acknowledges this case, handle the loading state within the component.
  const currentUserPending = state.pending[FETCH_FOR_CURRENT_USER] ||
    state.pending[FETCH_FOR_CURRENT_USER] === undefined
  const routeParams = { context: 'all', ...props.match.params }

  return {
    currentUser: getMe(state),
    currentUserPending,
    currentGroupMembership,
    group,
    groupPending: state.pending[FETCH_FOR_GROUP],
    hasMemberships,
    isDrawerOpen: get('AuthLayoutRouter.isDrawerOpen', state),
    isGroupMenuOpen: get('AuthLayoutRouter.isGroupMenuOpen', state),
    isGroupRoute: isGroupRoute(state, props),
    lastViewedGroup,
    returnToPath: getReturnToPath(state),
    routeParams,
    showMenuBadge,
    signupInProgress,
    slug
  }
}

export const mapDispatchToProps = {
  fetchForCurrentUser,
  fetchForGroup,
  setReturnToPath,
  toggleDrawer,
  toggleGroupMenu,
  updateUserSettings
}

export default connect(mapStateToProps, mapDispatchToProps)
