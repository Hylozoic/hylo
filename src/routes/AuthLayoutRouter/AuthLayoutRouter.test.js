import React from 'react'
import { shallow } from 'enzyme'
import AuthLayoutRouter from './AuthLayoutRouter'

it('shows NotFound if a currentUser is loaded and the group does not exist', () => {
  const wrapper = shallow(
    <AuthLayoutRouter
      isGroupRoute
      currentUser={{ hasRegistered: true }}
      location={{ pathname: '', search: '' }}
      routeParams={{ view: '' }}
    />
    , { disableLifecycleMethods: true }
  )
  expect(wrapper).toMatchSnapshot()
})

it('shows nothing for a group route if the group and currentUser are not loaded', () => {
  const wrapper = shallow(
    <AuthLayoutRouter
      currentUserPending
      isGroupRoute
      groupPending
      location={{ pathname: '/', search: '' }}
      routeParams={{ view: '' }}
    />
    , { disableLifecycleMethods: true }
  )
  expect(wrapper).toMatchSnapshot()
})

it('shows normal children for a group route if the group is loaded', () => {
  const wrapper = shallow(
    <AuthLayoutRouter
      currentUser={{ name: 'Testy Face', hasRegistered: true }}
      isGroupRoute
      group={{ id: '1' }}
      location={{ pathname: '/', search: '' }}
      routeParams={{ view: '' }}
    />
    , { disableLifecycleMethods: true }
  )
  expect(wrapper.name()).toEqual('Div100vh')
})

// WIP -- React Testing Library Tests
//
// Currently paused at mocking Intercom...

// import { Route } from 'react-router'
// import { graphql } from 'msw'
// import orm from 'store/models'
// import mockGraphqlServer from 'util/testing/mockGraphqlServer'
// import extractModelsForTest from 'util/testing/extractModelsForTest'
// import { AllTheProviders, render, screen } from 'util/testing/reactTestingLibraryExtended'
//
// jest.mock('store/selectors/getMixpanel', () => () => ({
//   identify: jest.fn(),
//   track: jest.fn(),
//   people: {
//     set: jest.fn()
//   }
// }))

// jest.mock('react-intercom', () => ({
//   IntercomAPI: () => jest.fn()
// }))

// jest.mock('client/rollbar', () => ({
//   error: jest.fn(),
//   configure: jest.fn()
// }))

// beforeEach(() => {
//   delete window.ResizeObserver
//   window.ResizeObserver = jest.fn().mockImplementation(() => ({
//     observe: jest.fn(),
//     unobserve: jest.fn(),
//     disconnect: jest.fn()
//   }))
// })

// // afterEach(() => {
// //   window.ResizeObserver = ResizeObserver
// //   jest.restoreAllMocks()
// // })

// function currentUserProvider (signupStateComplete) {
//   const ormSession = orm.mutableSession(orm.getEmptyState())
//   const reduxState = { orm: ormSession.state }

//   extractModelsForTest({
//     me: {
//       id: '1',
//       name: 'Test User',
//       hasRegistered: true,
//       emailValidated: true,
//       settings: {
//         signupInProgress: !signupStateComplete,
//         alreadySeenTour: true
//       },
//       memberships: [
//         {
//           id: '2',
//           person: {
//             id: '1'
//           },
//           group: {
//             id: '3',
//             slug: 'test-group'
//           }
//         }
//       ]
//     }
//   }, 'Me', ormSession)

//   return AllTheProviders(reduxState)
// }

// it('joins and forwards to group when current user is fully signed-up', async () => {
//   const me = {
//     id: '1',
//     name: 'Test User',
//     hasRegistered: true,
//     emailValidated: true,
//     settings: {
//       signupInProgress: false,
//       alreadySeenTour: true
//     },
//     memberships: [
//       {
//         id: '2',
//         person: {
//           id: '2'
//         },
//         group: {
//           id: '3',
//           slug: 'test-group',
//           name: 'Test Group'
//         }
//       }
//     ]
//   }

//   mockGraphqlServer.resetHandlers(
//     graphql.query('MeQuery', (req, res, ctx) => {
//       return res(
//         ctx.data({ me })
//       )
//     }),
//     graphql.query('FetchForGroup', (req, res, ctx) => {
//       return res(
//         ctx.data({ group: me.memberships[0].group })
//       )
//     })
//   )

//   await render(
//     <AuthLayoutRouter
//       location={{ pathname: '/groups/test-group', search: '' }}
//       match={{
//         params: {
//           groupSlug: 'test-group'
//         }
//       }}
//     />,
//     { wrapper: currentUserProvider(true) }
//   )
//   await screen.debug()
//   expect(await screen.findByText('Test Group')).toBeInTheDocument()
// })
