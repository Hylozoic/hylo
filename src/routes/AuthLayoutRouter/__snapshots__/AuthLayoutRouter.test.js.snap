// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows NotFound if a currentUser is loaded and the group does not exist 1`] = `
<Div100vh
  data-stylename="container"
>
  <RedirectRoute
    exact={true}
    key="/:context(public|all)/p/:postId"
    path="/:context(public|all)/p/:postId"
    to="/:context/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/:context(public|all)/project"
    path="/:context(public|all)/project"
    to="/:context/projects"
  />
  <RedirectRoute
    exact={true}
    key="/:context(public|all)/event"
    path="/:context(public|all)/event"
    to="/:context/events"
  />
  <RedirectRoute
    exact={true}
    key="/p/:postId"
    path="/p/:postId"
    to="/all/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/m/:personId"
    path="/m/:personId"
    to="/all/members/:personId"
  />
  <RedirectRoute
    exact={true}
    key="/m/:personId/p/:postId"
    path="/m/:personId/p/:postId"
    to="/all/members/:personId/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/all/m/:personId"
    path="/all/m/:personId"
    to="/all/members/:personId"
  />
  <RedirectRoute
    exact={true}
    key="/all/m/:personId/p/:postId"
    path="/all/m/:personId/p/:postId"
    to="/all/members/:personId/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/join/:accessCode"
    path="/(c|n)/:groupSlug/join/:accessCode"
    to="/groups/:groupSlug/join/:accessCode"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/"
    path="/(c|n)/:groupSlug/"
    to="/groups/:groupSlug/"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/event"
    path="/(c|n)/:groupSlug/event"
    to="/groups/:groupSlug/events"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/event/:postId"
    path="/(c|n)/:groupSlug/event/:postId"
    to="/groups/:groupSlug/events/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/project"
    path="/(c|n)/:groupSlug/project"
    to="/groups/:groupSlug/projects"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/project/:postId"
    path="/(c|n)/:groupSlug/project/:postId"
    to="/groups/:groupSlug/projects/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/:view(members|map|settings|topics)"
    path="/(c|n)/:groupSlug/:view(members|map|settings|topics)"
    to="/groups/:groupSlug/:view"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/map/p/:postId"
    path="/(c|n)/:groupSlug/map/p/:postId"
    to="/groups/:groupSlug/map/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/p/:postId"
    path="/(c|n)/:groupSlug/p/:postId"
    to="/groups/:groupSlug/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/groups/:groupSlug/p/:postId"
    path="/groups/:groupSlug/p/:postId"
    to="/groups/:groupSlug/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/m/:personId"
    path="/(c|n)/:groupSlug/m/:personId"
    to="/groups/:groupSlug/members/:personId"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/m/:personId/p/:postId"
    path="/(c|n)/:groupSlug/m/:personId/p/:postId"
    to="/groups/:groupSlug/members/:personId/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/:topicName"
    path="/(c|n)/:groupSlug/:topicName"
    to="/groups/:groupSlug/topics/:topicName"
  />
  <RedirectRoute
    exact={true}
    key="/(c|n)/:groupSlug/:topicName/p/:postId"
    path="/(c|n)/:groupSlug/:topicName/p/:postId"
    to="/groups/:groupSlug/topics/:topicName/post/:postId"
  />
  <RedirectRoute
    exact={true}
    key="/all/(members|settings)"
    path="/all/(members|settings)"
    to="/all"
  />
  <RedirectRoute
    exact={true}
    key="/public/(members|topics|settings)"
    path="/public/(members|topics|settings)"
    to="/public"
  />
  <Route
    path="/:context(all|public|groups)"
    render={[Function]}
  />
  <Route
    component={[Function]}
  />
  <Connect(TopNav)
    currentUser={
      Object {
        "hasRegistered": true,
      }
    }
    data-stylename="top"
    onClick={[Function]}
    routeParams={
      Object {
        "view": "",
      }
    }
  />
  <Route
    component={[Function]}
    path={
      Array [
        "/:context(groups)/:groupSlug/(.*)/create",
        "/:context(groups)/:groupSlug/create",
        "/:context(public|all)/(.*)/create",
        "/:context(public|all)/create",
        "(.*)/:detail(post)/:postId(\\\\d+)/:action(edit)",
      ]
    }
  />
  <div
    data-stylename="main"
    onClick={[Function]}
  >
    <Route
      component={[Function]}
      path={
        Array [
          "/:context(groups)/:groupSlug/:view?",
          "/:context(all|public)/:view?",
        ]
      }
    />
    <Div100vh
      data-stylename="center"
      id="center-column"
    >
      <Switch>
        <Route
          component={[Function]}
          path="/welcome"
        />
        <Route
          path="/:view(members)/:personId/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
          render={[Function]}
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(MemberProfile)",
              "type": [Function],
            }
          }
          path="/:context(all)/:view(members)/:personId/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Stream)",
              "type": [Function],
            }
          }
          path="/:context(all|public)/:view(stream)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Feed)",
              "type": [Function],
            }
          }
          path="/:context(all|public)/:view(projects)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Events)",
              "type": [Function],
            }
          }
          path="/:context(all|public)/:view(events)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(MapExplorer)",
              "type": [Function],
            }
          }
          path="/:context(all|public)/:view(map)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(MapExplorer)",
              "type": [Function],
            }
          }
          path="/:context(all|public)/:view(map)/:detail(group)?/(:detailGroupSlug)?"
        />
        <Route
          component={[Function]}
          path="/:context(public)/:view(groups)/:detail(group)?/(:detailGroupSlug)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Feed)",
              "type": [Function],
            }
          }
          path="/:context(all|public)/:view(topics)/:topicName"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(AllTopics)",
              "type": [Function],
            }
          }
          path="/:context(all)/:view(topics)"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Stream)",
              "type": [Function],
            }
          }
          path="/:context(all|public)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={[Function]}
          path={
            Array [
              "/:context(groups)/:groupSlug/join/:accessCode",
              "/h/use-invitation",
            ]
          }
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(MapExplorer)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(map)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(MapExplorer)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(map)/:detail(group)?/(:detailGroupSlug)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Stream)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(stream)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(LandingPage)",
              "type": [Function],
            }
          }
          exact={true}
          path="/:context(groups)/:groupSlug/:view(explore)/:detail(group)/:detailGroupSlug"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(LandingPage)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(explore)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Feed)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(projects)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Events)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(events)/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Groups)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(groups)"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Members)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(members)/create"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(MemberProfile)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(members)/:personId/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Members)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(members)"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Feed)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(topics)/:topicName/:detail(post)?/:postId(\\\\d+)?/:action(new|edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(AllTopics)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(topics)"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(GroupSettings)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug/:view(settings)"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Stream)",
              "type": [Function],
            }
          }
          exact={true}
          path="/:context(groups)/:groupSlug/:detail(post)/:postId(\\\\d+)/:action(edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Stream)",
              "type": [Function],
            }
          }
          path="/:context(groups)/:groupSlug"
        />
        <Route
          path="/messages/:messageThreadId?"
          render={[Function]}
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(UserSettings)",
              "type": [Function],
            }
          }
          path="/settings"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Search)",
              "type": [Function],
            }
          }
          path="/search"
        />
        <Redirect
          to="/all"
        />
      </Switch>
    </Div100vh>
    <div
      data-stylename="detail hidden"
      id="detail-column"
    >
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(PostDetail)",
              "type": [Function],
            }
          }
          key="/:context(all|public)/:view(events|explore|map|projects|stream)/:detail(post)/:postId(\\\\d+)/:action(edit)?"
          path="/:context(all|public)/:view(events|explore|map|projects|stream)/:detail(post)/:postId(\\\\d+)/:action(edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(GroupDetail)",
              "type": [Function],
            }
          }
          key="/:context(all|public)/:view(map)/:detail(group)/:detailGroupSlug"
          path="/:context(all|public)/:view(map)/:detail(group)/:detailGroupSlug"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(GroupDetail)",
              "type": [Function],
            }
          }
          key="/:context(public)/:view(groups)/:detail(group)/:detailGroupSlug"
          path="/:context(public)/:view(groups)/:detail(group)/:detailGroupSlug"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(PostDetail)",
              "type": [Function],
            }
          }
          key="/:context(all)/:view(members)/:personId/:detail(post)/:postId(\\\\d+)/:action(edit)?"
          path="/:context(all)/:view(members)/:personId/:detail(post)/:postId(\\\\d+)/:action(edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(PostDetail)",
              "type": [Function],
            }
          }
          key="/:context(all|public)/:detail(post)/:postId(\\\\d+)/:action(edit)?"
          path="/:context(all|public)/:detail(post)/:postId(\\\\d+)/:action(edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(PostDetail)",
              "type": [Function],
            }
          }
          key="/:context(groups)/:groupSlug/:view(events|explore|map|projects|stream)/:detail(post)/:postId(\\\\d+)/:action(edit)?"
          path="/:context(groups)/:groupSlug/:view(events|explore|map|projects|stream)/:detail(post)/:postId(\\\\d+)/:action(edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(PostDetail)",
              "type": [Function],
            }
          }
          key="/:context(groups)/:groupSlug/:view(members)/:personId/:detail(post)/:postId(\\\\d+)/:action(edit)?"
          path="/:context(groups)/:groupSlug/:view(members)/:personId/:detail(post)/:postId(\\\\d+)/:action(edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(GroupDetail)",
              "type": [Function],
            }
          }
          key="/:context(groups)/:groupSlug/:view(explore|map|groups)/:detail(group)/:detailGroupSlug"
          path="/:context(groups)/:groupSlug/:view(explore|map|groups)/:detail(group)/:detailGroupSlug"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(PostDetail)",
              "type": [Function],
            }
          }
          key="/:context(groups)/:groupSlug/:view(topics)/:topicName/:detail(post)/:postId(\\\\d+)/:action(edit)?"
          path="/:context(groups)/:groupSlug/:view(topics)/:topicName/:detail(post)/:postId(\\\\d+)/:action(edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(PostDetail)",
              "type": [Function],
            }
          }
          key="/:context(groups)/:groupSlug/:detail(post)/:postId(\\\\d+)/:action(edit)?"
          path="/:context(groups)/:groupSlug/:detail(post)/:postId(\\\\d+)/:action(edit)?"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(GroupDetail)",
              "type": [Function],
            }
          }
          key="/:context(groups)/:groupSlug/:detail(group)/:detailGroupSlug"
          path="/:context(groups)/:groupSlug/:detail(group)/:detailGroupSlug"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(PostDetail)",
              "type": [Function],
            }
          }
          key="/:view(members)/:personId/:detail(post)/:postId(\\\\d+)/:action(edit)?"
          path="/:view(members)/:personId/:detail(post)/:postId(\\\\d+)/:action(edit)?"
        />
      </Switch>
    </div>
  </div>
  <Connect(SocketListener)
    location={
      Object {
        "pathname": "",
        "search": "",
      }
    }
  />
  <Connect(SocketSubscriber)
    type="group"
  />
  <Intercom
    appID=""
    hide_default_launcher={true}
  />
</Div100vh>
`;

exports[`shows nothing for a group route if the group and currentUser are not loaded 1`] = `
<div
  data-stylename="container"
>
  <Loading
    type="loading-fullscreen"
  />
</div>
`;
